pkg/kubelet/certificate/certificate_manager.go

```diff
@@ -46,85 +46,142 @@ const (
 // manager. In the background it communicates with the API server to get new
 // certificates for certificates about to expire.
 type Manager interface {
+	// CertificateSigningRequestClient sets the client interface that is used for
+	// signing new certificates generated as part of rotation.
+	CertificateSigningRequestClient(certificatesclient.CertificateSigningRequestInterface)
```

Comment:
It's an awkward situation alright. Since the manager uses the REST client to do rotations, I need the [REST client to initialize the manager](https://github.com/jcbsmpsn/kubernetes/blob/rotate-client-certificate/cmd/kubelet/app/server.go#L419), and I need the [manager to supply the keys](https://github.com/jcbsmpsn/kubernetes/blob/rotate-client-certificate/cmd/kubelet/app/server.go#L409-L411) to initialize the REST client.